### ISSUE TITLE
state_dict error when training Blender Bot 2.0

### ISSUE THREAD

**Issue by miketcchung** [facebookresearch/ParlAI#3983]
I'm trying to train the Blender Bot 2.0 400M and 2.7B model on Google Colab with a custom dataset but it's giving me a state_dict error. 
![BlenderBot 2 0 state_dict error](https://user-images.githubusercontent.com/48034150/131019648-1c0b9068-5374-4016-87eb-04d346a3cc1e.png)
Here is the code I'm using to train the model:

`from parlai.scripts.train_model import TrainModel

TrainModel.main(
    # similar to before
    task='fromfile:parlaiformat', 
    fromfile_datapath='/content/gdrive/My Drive/Athlete_Coach_Bot/fitness-faq-parlai.txt',
    model='projects.blenderbot2.agents.blenderbot2:BlenderBot2FidAgent',
    model_file='model_finetune/model',
    
    # initialize with a pretrained model
    init_model='zoo:blenderbot2/blenderbot2_3B/model',
    
    # arguments we get from the pretrained model.
    # Unfortunately, these must be looked up separately for each model.
    # model/neural net info
    n_layers=2, ffn_size=10240, dropout=0.0, attention_dropout=0.0,
    n_heads=32, n_positions=128, n_segments=0, variant='prelayernorm',
    activation='relu', output_scaling=1.0, memory_attention='sqrt',
    reduction_type='mean', 

    # embdding info
    retriever_embedding_size=768, learn_embeddings=True,
    embedding_type='random', share_word_embeddings=True,
    embeddings_scale=True, embedding_size=2560,
    embedding_projection='random', learn_positional_embeddings=True,

    # dictionary info/pre-processing
    dict_minfreq=0, dict_nulltoken='__null__', dict_maxtokens=-1,
    dict_lower=False, cap_num_predictions=100, dict_textfields='text,labels',
    dict_language='english', dict_tokenizer='bytelevelbpe', 
    dict_max_ngram_size=-1, dict_endtoken='__end__', train_predict=False, 
    dict_unktoken='__unk__',  dict_class='parlai.core.dict:DictionaryAgent', 
    dict_starttoken='__start__', 
    dict_file='zoo:blenderbot2/blenderbot2_3B/model.dict',

    # other dataset-related info
    truncate=128, text_truncate=128, label_truncate=128, split_lines=False,
    
    # some training arguments, specific to this fine-tuning
    batchsize=16, lr=1e-05, lr_scheduler='reduceonplateau', 
    multitask_weights=[3.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0], 
    validation_metric='ppl', max_train_time=600, validation_every_n_epochs=0.25,

    # more batch and learning rate info
    lr_scheduler_decay=0.5, lr_scheduler_patience=3, invsqrt_lr_decay_gamma=-1,
    encode_candidate_vecs_batchsize=256,

    # training info
    gpu=-1, data_parallel=False, optimizer='mem_eff_adam', gradient_clip=0.1,
    adam_eps=1e-08, nesterov=True, nus=[0.7], betas=[0.9, 0.999],
    warmup_updates=100, warmup_rate=0.0001, update_freq=1, fp16=True,
    no_cuda=False,

    # speeds up validation
    skip_generation=True,
    
    # helps us cram more examples into our gpu at a time
    dynamic_batching='full',
)`

I copied the hyperparameters from the model card https://github.com/facebookresearch/ParlAI/blob/master/parlai/zoo/blenderbot2/model_card.md. What's going wrong? Is there some example code for how to train Blender Bot 2.0 with a custom dataset?

**Comment by klshuster** [comment]
hmm, a few parameters seemed to have dropped from the model card; you'll want to set the following:

`n_encoder_layers=2, n_decoder_layers=24`

**Comment by mdsaifk** [comment]
hi @miketcchung are you able to train the custom model using blender Bot , if yes please share the instructions
